src/IO/WriteBufferFromEncryptedFile.cpp

```diff
@@ -21,7 +22,9 @@ WriteBufferFromEncryptedFile::WriteBufferFromEncryptedFile(
 
 WriteBufferFromEncryptedFile::~WriteBufferFromEncryptedFile()
 {
-    finalize();
```

Comment:
`finalize()` must be called before calling the destructor of any write buffer.
And `~WriteBuffer()` even has the [special code](https://github.com/ClickHouse/ClickHouse/blob/2fd0f7be7b9765e28d5cb78bb8d3c2725f2ede2f/src/IO/WriteBuffer.cpp#L13-L30) to check that `finalize()` is called.

Also it's perfectly all right not to call `finalize()` before calling the destructor in case of exception.